### DISCUSSION TITLE
While running with splunk-otel-agent 1.2, can API 1.4.1 create custom spans via Tracer?

### DISCUSSION THREAD

**Discussion by woldie** [open-telemetry/opentelemetry-java#3531] [Q&A]
I am auto-instrumenting with the splunk-otel-agent 1.2, and I'd like to create custom spans using the same Tracer as the one the agent uses.   I am currently attempting to acquire the Tracer with API 1.4.1:

```
GlobalOpenTelemetry.getTracer(Objects.requireNonNull(applicationContext.getId()))
```

And I am trying to create custom spans with

```
tracer.spanBuilder("mySpan")
        .setSpanKind(SpanKind.SERVER)
        .startSpan();
```

Unfortunately, my custom spans are not traced, and only the auto-instrumented spans the agent creates show up in my traces in SignalFX.  I am calling Span#end on all my custom spans.

This link referenced in another similar answer is no longer relevant:  https://github.com/open-telemetry/opentelemetry-java/blob/main/QUICKSTART.md#tracing

I would like to know the following:

1) Is it possible to access/connect to the same Tracer that splunk-otel-agent must have internally?
2) If yes, is it possible (without using @WithSpan) to add custom spans to my service via that Tracer?

Thanks!

**Comment by anuraaga** [comment]
Unfortunately, this is not supported. Using a newer version of the agent with older API is, the other way around often won't work.  This is because the agent is the one in control - it knows about older versions and can act accordingly but there is no way for it to be ready for a new version it doesn't know about. It may still work but no guarantees.

The solution is to update the agent which hopefully isn't too hard since the app doesn't need to be rebuilt.

**Reply by jkwatson** [reply]
According to the release notes, this should be the correct combination : https://github.com/signalfx/splunk-otel-java/releases/tag/v1.2.0

I think this is probably a better question for the splunk-otel-java repository than here. Do you mind asking over there where we can get all the splunk folks involved?

**Reply by jkwatson** [reply]
I wonder if this is being repressed due to nested SERVER span logic

**Reply by anuraaga** [reply]
Sorry for the fake note, didn't realize Splunk is using a different versioning scheme!

**Comment by woldie** [comment]
Oh, should I only use CLIENT span kinds?  I can try that.  I will also
reach out to the Splunk folks and ask there.

Thanks for responding, all!

**Reply by jkwatson** [reply]
keeping them as INTERNAL is also just fine. You don't need to set a span kind if you just want a standard INTERNAL span.
